mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -61,20 +67,70 @@ class LogisticRegressionModel (
 
   override protected def predictPoint(dataMatrix: Vector, weightMatrix: Vector,
```

Comment:
I thought about having weights as a matrix, but it's required to change so many places. For example, the gradient object has to change, the underline `GeneralizedLinearAlgorithm` has to change as well. I'm thinking to have more clear APIs when we move the code to `ml` package since we can do it from scratch.